single-pool/js/tests/transactions.test.ts

```diff
@@ -0,0 +1,385 @@
+  let userStakeAccount = new Keypair();
+
+  const stakeRent = await connection.getMinimumBalanceForRentExemption(StakeProgram.space);
+  const minimumDelegation = (await connection.getStakeMinimumDelegation()).value;
+  let transaction = StakeProgram.createAccount({
+    authorized: new Authorized(context.payer.publicKey, context.payer.publicKey),
+    fromPubkey: context.payer.publicKey,
+    lamports: stakeRent + minimumDelegation,
+    stakePubkey: userStakeAccount.publicKey,
+  });
+  await processTransaction(context, transaction, [userStakeAccount]);
+  userStakeAccount = userStakeAccount.publicKey;
+
+  transaction = StakeProgram.delegate({
+    authorizedPubkey: context.payer.publicKey,
+    stakePubkey: userStakeAccount,
+    votePubkey: voteAccountAddress,
+  });
+  await processTransaction(context, transaction);
+
+  return userStakeAccount;
+}
+
+test('initialize', async (t) => {
+  const context = await startWithContext();
+  const client = context.banksClient;
+  const payer = context.payer;
+  const connection = new BanksConnection(client, payer);
+
+  const voteAccountAddress = new PublicKey(voteAccount.pubkey);
+  const poolAddress = findPoolAddress(SinglePoolProgram.programId, voteAccountAddress);
+
+  // initialize pool
+  const transaction = await SinglePoolProgram.initialize(
+    connection,
+    voteAccountAddress,
+    payer.publicKey,
+  );
+  await processTransaction(context, transaction);
+
+  t.truthy(await client.getAccount(poolAddress), 'pool has been created');
+  t.truthy(
+    await client.getAccount(
+      findMplMetadataAddress(findPoolMintAddress(SinglePoolProgram.programId, poolAddress)),
+    ),
+    'metadata has been created',
+  );
+});
+
+test('deposit', async (t) => {
+  const context = await startWithContext();
+  const client = context.banksClient;
+  const payer = context.payer;
+  const connection = new BanksConnection(client, payer);
+
+  const voteAccountAddress = new PublicKey(voteAccount.pubkey);
+  const poolAddress = findPoolAddress(SinglePoolProgram.programId, voteAccountAddress);
+  const poolStakeAddress = findPoolStakeAddress(SinglePoolProgram.programId, poolAddress);
+  const userStakeAccount = await createAndDelegateStakeAccount(context, voteAccountAddress);
+
+  // initialize pool
+  let transaction = await SinglePoolProgram.initialize(
+    connection,
+    voteAccountAddress,
+    payer.publicKey,
+  );
+  await processTransaction(context, transaction);
+
+  // deposit
+  transaction = await SinglePoolProgram.deposit({
+    connection,
+    pool: poolAddress,
+    userWallet: payer.publicKey,
+    userStakeAccount,
+  });
+  await processTransaction(context, transaction);
+
+  const minimumDelegation = (await connection.getStakeMinimumDelegation()).value;
+  const poolStakeAccount = await client.getAccount(poolStakeAddress);
+  t.true(poolStakeAccount.lamports > minimumDelegation * 2, 'stake has been deposited');
```

Comment:
up to you, but it might be nice to include the rent exemption and make this a full `===` test

Reply:
i thought it would be potentially brittle because the amount depends on whether the pool is activating or active at time of deposit, but i think bankrun lets me jump epochs

Reply:
that's fair, it's up to you!